server/mscalendar/notification_test.go

```diff
@@ -0,0 +1,190 @@
+			Response: "event_response",
+		},
+		Weblink:           "event_weblink",
+		Subject:           "event_subject",
+		BodyPreview:       "event_body_preview",
+		ResponseRequested: true,
+	}
+
+	sStore := &store.Subscription{
+		PluginVersion: "x.x.x",
+		Remote: &remote.Subscription{
+			ID:          "remote_subscription_id",
+			ClientState: "stored_client_state",
+		},
+		MattermostCreatorID: "creator_mm_id",
+	}
+
+	uStore := &store.User{
+		Settings: store.Settings{
+			EventSubscriptionID: "remote_subscription_id",
+		},
+		Remote:           &remote.User{},
+		OAuth2Token:      token,
+		MattermostUserID: "creator_mm_id",
+	}
+
+	nRemote := &remote.Notification{
+		SubscriptionID: "remote_subscription_id",
+		IsBare:         true,
+		SubscriptionCreator: &remote.User{},
+		Event:               &remote.Event{},
+		Subscription:        &remote.Subscription{},
+	}
+
+	tcs := []struct {
+		name                string
+		incomingClientState string
+		expectedError       string
+		hasPriorEvent       bool
+		recommendRenew      bool
+	}{
+		{
+			name:                "incoming ClientState matches stored ClientState",
+			incomingClientState: "stored_client_state",
+			expectedError:       "",
+			hasPriorEvent:       false,
+			recommendRenew:      false,
+		}, {
+			name:                "incoming ClientState doesn't match stored ClientState",
+			incomingClientState: "wrong_client_state",
+			expectedError:       "Unauthorized webhook",
+			hasPriorEvent:       false,
+			recommendRenew:      false,
+		}, {
+			name:                "prior event exists",
+			incomingClientState: "stored_client_state",
+			expectedError:       "",
+			hasPriorEvent:       true,
+			recommendRenew:      false,
+		}, {
+			name:                "sub renewal recommended",
+			incomingClientState: "stored_client_state",
+			expectedError:       "",
+			hasPriorEvent:       false,
+			recommendRenew:      true,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := mock_store.NewMockStore(ctrl)
+			logger := &bot.TestLogger{TB: t}
+			mockPoster := mock_bot.NewMockPoster(ctrl)
+			mockRemote := mock_remote.NewMockRemote(ctrl)
+			mockPluginAPI := mock_plugin_api.NewMockPluginAPI(ctrl)
+			mockClient := mock_remote.NewMockClient(ctrl)
+
+			conf := &config.Config{BotUserID: "bot_mm_id", PluginVersion: "x.x.x"}
+			env := Env{
+				Config: conf,
+				Dependencies: &Dependencies{
+					Store:     mockStore,
+					Logger:    logger,
+					Poster:    mockPoster,
+					Remote:    mockRemote,
+					PluginAPI: mockPluginAPI,
+				},
+			}
+
+			nRemote.ClientState = tc.incomingClientState
+			nRemote.RecommendRenew = tc.recommendRenew
+
+			nRemoteWithEvent := *nRemote
+			nRemoteWithEvent.Event = eRemote
+
+			mockStore.EXPECT().LoadSubscription("remote_subscription_id").Return(sStore, nil).Times(1)
+			mockStore.EXPECT().LoadUser("creator_mm_id").Return(uStore, nil).Times(1)
```

Comment:
The name `uStore` makes me think the variable is an instance of `UserStore`. 1/5 I think the variable name `user` is sufficient. Other similar variable names should probably be changed as well.